src/main/java/net/neoforged/neoforge/common/util/BlockSnapshot.java

```diff
@@ -120,18 +120,29 @@ public boolean restoreToLocation(LevelAccessor world, BlockPos pos, boolean forc
         if (world instanceof Level)
             ((Level) world).sendBlockUpdated(pos, current, replaced, flags);
 
+        restoreBlockEntity(world, pos);
+
+        if (DEBUG)
+            System.out.println("Restored " + this.toString());
+        return true;
+    }
+
+    /**
+     * Loads the stored {@link BlockEntity} data if one exists at the given position.
+     * 
+     * @return true if any data was loaded
+     */
+    public boolean restoreBlockEntity(LevelAccessor world, BlockPos pos) {
         BlockEntity te = null;
```

Comment:
Since we're here anyway, it may be worth changing these variable names to match the renaming of TileEntity to BlockEntity

Reply:
Hornet's nest you've poked there :p

Did a full cleanup pass on that class